include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -586,6 +586,35 @@ def TTNN_SliceOp: TTNN_NamedDPSOp<"slice"> {
     let hasVerifier = 1;
 }
 
+def TTNN_LinearOp : TTNN_NamedDPSOp<"linear"> {
+    let summary = "Linear transformation of inputs.";
+
+    let description = [{
+      Produces the matmul of tensors `a` and `b` with optional addition with `bias`.
+
+      Example:
+        // %a = [[1., 2.]], [2., 1.]]
+        // %b = [[0., 1.], [1., 0.]]
+        // %bias = [[1.]]
+        "ttnn.linear"(%a, %b, %bias, %result) : (tensor<2x2xf16>, tensor<2x2xf16>, tensor<1xf16>, tensor<2x2xf16>) -> tensor<2x2xf16>
+        // %result = [[3., 2.], [2., 3.]]
+    }];
+
+    let arguments = (ins AnyRankedTensor:$a,
+                         AnyRankedTensor:$b,
+                         Optional<AnyRankedTensor>:$bias,
+                         AnyRankedTensor:$output);
```

Comment:
we don't need dps for linear in ttnn nor do we need it for matmul. 
Can you fix this for linear are create item to remove it from matmul.

Reply:
It's WIP to make matmul (and by extension linear) DPS on in TTNN https://github.com/tenstorrent/tt-metal/issues/15038. I would leave it as DPS in dialect if the change is going to land in the near feature, because it would require non trivial modifications on our side for both linear and matmul.